src/dsql/WinNodes.cpp

```diff
@@ -432,7 +432,7 @@ void FirstValueWinNode::aggInit(thread_db* tdbb, Request* request) const
 
 dsc* FirstValueWinNode::winPass(thread_db* tdbb, Request* request, SlidingWindow* window) const
 {
-	if (!window->moveWithinFrame(-(window->getRecordPosition() - window->getFrameStart())))
+	if (!window->moveWithinFrame(-static_cast<SINT64>(window->getRecordPosition() - window->getFrameStart())))
```

Comment:
I\d suggest a temporary variable for better readability, e.g.:
```C++
const SINT64 offset = window->getRecordPosition() - window->getFrameStart(); // with static_cast, if necessary
if (!window->moveWithinFrame(-offset))
```

Reply:
Expression  `window->getRecordPosition() - window->getFrameStart()` used many times, makes sence to move it into separate function, like:

```
	class SlidingWindow
	{
	public:
...
	FB_UINT64 getInFrameOffset() const
	{
		return savedPosition - frameStart;
	}

```